sync/validation_queue.go

```diff
@@ -101,9 +101,25 @@ func (vq *validationQueue) traverse(s *Syncer, blk *types.BlockHeader) error {
 func (vq *validationQueue) finishBlockCallback(s *Syncer, block *types.Block) func() error {
 	return func() error {
 		//data availability
-		txs, atxs, err := s.DataAvailabilty(block)
+		atxsHash12, atxsRes, atxsSeen, err := s.checkAtxCache(block)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to check ATXs cache - err %v", err)
+		}
+		if atxsSeen == true && atxsRes == false {
+			return fmt.Errorf("invalid atxs according to cache (key %v)", atxsHash12)
+		}
+		txs, txErr, atxs, atxErr := s.DataAvailability(block, atxsSeen)
+		if txErr != nil || atxErr != nil {
+			if atxErr != nil {
+				// update cache only on failures, positive caching is done after all ATXs are processed
+				s.updateAtxCache(atxsHash12, false)
```

Comment:
im not sure why we want to fail fetching atxs that we failed to fetch in the past, idont think the failure to fetch necessarily suggests that they are faulty

Reply:
if you agree with me on this than we can just check the cache inside DataAvailabilty and avoid the code duplication here and in BlockSyntacticValidation

Reply:
per our discussion, removed both caches